### DISCUSSION TITLE
Vendored files suddenly being included in statistics (no changes on my end)

### DISCUSSION THREAD

**Discussion by malwaredllc** [github-linguist/linguist#5329] [Classification] [answered]
Repo: https://github.com/malwaredllc/byob
.gitattributes file: https://github.com/malwaredllc/byob/blob/master/.gitattributes

This has always been a Python project, and suddenly today it is showing as JavaScript despite no changes on my end.

Did something change with the linguist project? Is there a change I need to make in my .gitattributes files? Anyone know how to fix this?

**Comment by lildude** [comment, marked as answer]
This is the second time this has come up this week. Nothing has changed in Linguist that would cause this so I've dug into it and I've found that we've recently updated the version of the rugged gem to a much newer version that has pulled in fixes from libgit2 that has fixed the old incorrect behaviour bringing GitHub's behaviour inline with git's. I think https://github.com/libgit2/libgit2/pull/5110 is the specific PR.

As with the first case in https://github.com/github/linguist/discussions/5323, your override isn't quite right as it's not recursive so it shouldn't have worked correctly in the past. It certainly wouldn't work directly with git in the command line.

https://github.com/github/linguist/discussions/5175#discussioncomment-317382 and https://github.com/github/linguist/discussions/5308#discussioncomment-561606 provide more info on testing the overrides directly with git which GitHub should now match.

**Reply by malwaredllc** [reply]
Thanks for the quick response! Based on your notes, I was able to fix the issue by making the override recursive.
